lib/server.js

```diff
@@ -96,7 +103,16 @@ class Server extends KarmaEventEmitter {
   }
 
   start () {
-    this._injector.invoke(this._start, this)
+    return Promise.all([
+      BundleUtils.bundleResourceIfNotExist('client/main.js', 'static/karma.js'),
+      BundleUtils.bundleResourceIfNotExist('context/main.js', 'static/context.js')
+    ])
+      .then(() => NetUtils.getAvailablePort(this.get('config.port'), this.get('config.listenAddress')))
```

Comment:
does it reasonable to get `config` and reuse it below instead of getting it each time?
```js
const config = this.get('config')
...
.then(() => NetUtils.getAvailablePort(config.port, config.listenAddress))
...
config.port = port
```

Reply:
Updated